node/src/server-modules/GlideFt.ts

```diff
@@ -269,6 +269,124 @@ export class GlideFt {
             >
         ).then(convertGlideRecordToRecord);
     }
+
+    /**
+     * Uses the provided query expression to locate keys within an index. Once located, the count
+     * and/or content of indexed fields within those keys can be returned.
+     *
+     * @param client The client to execute the command.
+     * @param indexName The index name to search into.
+     * @param query The text query to search.
+     * @param options - (Optional) See {@link FTSearchOptions} and {@link DecoderOption}.
+     *
+     * @returns A two-element array, where the first element is number of documents in the result set, and the
+     * second element, has the format: `GlideRecord<GlideString | GlideRecord<GlideString>>`:
+     * a mapping between document names and a map of their attributes.
+     *
+     * If `count` or `limit` with values `{offset: 0, count: 0}` is
+     * set, the command returns array with only one element: the number of documents.
+     *
+     * @example
+     * ```typescript
+     * //
+     * const vector = Buffer.alloc(24);
+     * const result = await GlideFt.search(client, "json_idx1", "*=>[KNN 2 @VEC $query_vec]", {params: [{key: "query_vec", value: vector}]});
+     * console.log(result); // Output:
+     * // {
+     * //   2,
+     * //   [
+     * //     {
+     * //       "key": "json:2",
+     * //       "value":[
+     * //         {
+     * //           "key": "$",
+     * //           "value": "{\"vec\":[1.1,1.2,1.3,1.4,1.5,1.6]}",
+     * //         },
+     * //         {
+     * //           "key": "__VEC_score",
+     * //           "value": "11.1100006104",
+     * //         }
+     * //       ]
+     * //     },
+     * //     {
+     * //       "key": "json:0",
+     * //       "value":[
+     * //         {
+     * //           "key": "$",
+     * //           "value": "{\"vec\":[1,2,3,4,5,6]}",
+     * //         },
+     * //         {
+     * //           "key": "__VEC_score",
+     * //           "value": "91",
+     * //         }
+     * //       ]
+     * //     }
+     * //   ]
+     * // }
+     * ```
+     */
+    static async search(
+        client: GlideClient | GlideClusterClient,
+        indexName: GlideString,
+        query: GlideString,
+        options?: FTSearchOptions & DecoderOption,
+    ): Promise<
+        (number | GlideRecord<GlideString | GlideRecord<GlideString>>)[]
+    > {
+        const args: GlideString[] = ["FT.SEARCH", indexName, query];
+
+        if (options) {
```

Comment:
It is possible to move this code to FtSearchOptions and call a function options.toArgs()?

Reply:
`toArgs` is actually bad.  The function is internal that we have to expose publicly on other clients.  The best way to do this is to over engineer a solution that includes the public model (the options class), a private converter or mapper (including the toArgs function) and this command function that calls the private mapper.  

I think that's over doing it.  But in this case, it makes more sense to hide the business logic from the public interface.